src/parser/serializer.ts

```diff
@@ -767,7 +767,7 @@ export function serializeInto(
       let value = object[i];
 
       // Is there an override value
-      if (value && value.toBSON) {
+      if (value !== null && value !== undefined && value.toBSON) {
         if (typeof value.toBSON !== 'function') throw new BSONTypeError('toBSON is not a function');
         value = value.toBSON();
       }
```

Comment:
```suggestion
      if (typeof value?.toBSON === 'function') {
        value = value.toBSON();
      }
```

We can use the optional chaining syntax here to be a little more succinct, null and undefined will not be accessed with a dot so `typeof` will return `'undefined'` so the if statement won't be entered.

This is what it compiles to, just so we can all double check this is what's expected.
```js
typeof (value === null || value === undefined ? undefined : value.toBSON) === 'function'
```